Project Euler 57 - Square root convergents

bh2smith opened the pull request:
### **Describe your change:**

This PR addresses issue #2695 and adds a solution to Project Euler Problem 57 - "Square root convergents"


* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory. 
**Created new directory for previously non-existent Project Euler solution**
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
**Doesn't seem relevant**
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

dhruvmanila reviewed (commented):
### Announcement:

@bh2smith

*This message is sent to all pull requests which are **OPEN, NOT MERGED, not containing 'wontfix' label and contains 'Project Euler' or 'Project' or 'Euler' (case doesn't matter) in their title**. If this message doesn't apply to your pull request, please ignore this.*

### Message:

This is to notify all the users submitting a pull request for the Project Euler solution that the [README.md](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) file has been updated with all the style requirements necessary for the directory and solution files. ***Please read through it and make all the necessary changes required to your submission.***

bh2smith commented:
Thanks @dhruvmanila for the feedback, I have updated the solution to comply with the new guidelines. In particular, 

- Renamed directory to have 3 digits for problem number.
- Included link to the original problem and 
- removed the value for input in the main method.

Please let me know if anything else looks off! Cheers 🥂

dhruvmanila commented on project_euler/problem_057/sol1.py:
```diff
@@ -0,0 +1,48 @@
+    >>> solution(10000)
+    1508
+    """
+    a, b = 1, 1
+    res = []
```
Descriptive variable names would be awesome to work with :)

bh2smith replied:
Great idea. I have made the replacements 

```
a -> prev_numerator
b -> prev_denominator
res -> result
```

dhruvmanila reviewed (approved):
Thank you for your contribution!

bh2smith commented:
> Thank you for your contribution!

I've got plenty more coming your way! Also hope to contribute to the Rust Project Euler Solutions as well.
